### ISSUE TITLE
PRECISION=SINGLE will not work

### ISSUE THREAD

**Issue by zingale** [BoxLib-Codes/Castro#34]
Formally, boxlib supports double and single precision, and can be compiled to support either via the `PRECISION` variable in the makefiles.

However, all of the Fortran routines that we write assume that we pass data through as `double *`, so only double precision will work.

**Comment by WeiqunZhang** [comment]
This is actually something I am working on now.  I am adding a fortran module to C_BaseLib.  It includes a c_real type which is C++ BoxLib's Real.  On fortran side, we can then use real(c_real) x.

**Comment by WeiqunZhang** [comment]
The bl_fort_module is in BoxLib now.  Note that it also contains an integer parameter, bl_spacedim.

```
module bl_fort_module

  use iso_c_binding, only : c_float, c_double

  implicit none

  integer, parameter :: bl_spacedim = BL_SPACEDIM

#ifdef BL_USE_FLOAT
  integer, parameter :: c_real = c_float
#else
  integer, parameter :: c_real = c_double
#endif

end module bl_fort_module
```

**Comment by zingale** [comment]
should we change the Castro Fortran code to use `real(c_real) :: x` now, or should we not worry about this.  It is unlikely that we will ever get the whole code, e.g., Microphysics, working in single precision.  Also unclear if this is desired.  

An alternate _fix_ is to just abort in the make if someone tries to compile in SINGLE precision.

**Comment by maxpkatz** [comment]
We should change the Fortrn code to use real(c_real) independent of this change. But that would indirectly solve much of this problem.

**Comment by zingale** [comment]
so does this mean that any constants in the code should then be like

```
1.0_c_real
```

**Comment by zingale** [comment]
started a script in `Castro/util/scripts/dpconvert.py` that will make this change.  It does the declarations at the moment, but still needs to address constants and adding the use statement in the appropriate place.

**Comment by zingale** [comment]
this is mostly addressed by PR #88
